metadata/src/main/java/org/apache/kafka/controller/DelegationTokenControlManager.java

```diff
@@ -341,9 +345,12 @@ public List<ApiMessageAndVersion> sweepExpiredDelegationTokens() {
                     oldTokenInformation.tokenId(), oldTokenInformation.ownerAsString());
                 records.add(new ApiMessageAndVersion(new RemoveDelegationTokenRecord().
                     setTokenId(oldTokenInformation.tokenId()), (short) 0));
+                if (records.size() >= MAX_RECORDS_PER_EXPIRATION) {
```

Comment:
Why limit this to 1000? Just to avoid batch size issues?

Reply:
batch size issues, and also to avoid high latencies.